specification/http/1.0/openapi.json

```diff
@@ -0,0 +1,456 @@
+              "type": "string",
+              "description": "The secret's value."
+            }
+          },
+          "required": ["id", "value"],
+          "additionalProperties": false
+        },
+        "SimpleJSONSchema": {
+          "type": "object",
+          "properties": {
+            "type": "object",
+            "additionalProperties": true,
+            "not": {
+              "properties": {
+                "$ref": {},
+                "$defs": {},
+                "definitions": {}
+              }
+            }
+          }
+        },
+        "ToolDefinition": {
+          "type": "object",
+          "properties": {
+            "id": {
+              "$ref": "#/components/schemas/ToolId"
+            },
+            "name": {
+              "type": "string",
+              "description": "The tool's name."
+            },
+            "description": {
+              "type": "string",
+              "description": "A human-readable explanation of the tool's purpose. This field can be used by both humans and AI models."
+            },
+            "toolkit": {
+              "type": "object",
+              "properties": {
+                "name": {
+                  "type": "string",
+                  "description": "The name of the toolkit."
+                },
+                "description": {
+                  "type": "string",
+                  "description": "A human-readable description of the toolkit."
+                },
+                "version": {
+                  "type": "string",
+                  "description": "An identifier for this version of the toolkit."
+                }
+              },
+              "required": ["name", "description"],
+              "additionalProperties": false
+            },
+            "input": {
+              "type": "object",
+              "properties": {
+                "parameters": {
+                  "$ref": "#/components/schemas/SimpleJSONSchema",
+                  "description": "JSON Schema describing the input parameters for the tool. Supports standard JSON Schema validation but excludes $ref and definitions/schemas for simplicity.",
+                  "additionalProperties": true,
+                  "required": ["name", "description"]
+                },
+                "non_inferrable_parameters": {
+                  "type": "array",
+                  "description": "A list of parameter names that must not be inferred by a model.",
+                  "items": {
+                    "type": "string"
+                  }
+                }
+              },
+              "required": ["parameters"],
+              "additionalProperties": false
+            },
+            "output": {
+              "type": "object",
+              "properties": {
+                "available_modes": {
+                  "type": "array",
+                  "items": {
+                    "type": "string",
+                    "enum": ["value", "error", "null"]
```

Comment:
* What is "null"?
* Why not just say that in the spec tools should always be assumed to return a union of value , error and client implementation?

Reply:
I do agree this could be simplified to `value,error` or `none,error`. I think that would be a good change.